src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -46,6 +52,12 @@ public void addNode(T node) {
     m_nodesReady.add(node);
   }
 
+  public void addEdge(int weight, T from, T to, int order) {
+    List<Edge<T>> edges = Lists.newArrayList();
+    edges.add(new Edge<>(from, to, weight, order));
+    addEdges(edges);
```

Comment:
`addEdges(Collections.singletonList(new Edge<>(from, to, weight, order)));`

But you can extract and use a new method too: `private void addEdge(Edge<T> edge)` (in fact, my favorite solution)